src/neo/SmartContract/ApplicationEngine.Crypto.cs

```diff
@@ -14,9 +14,14 @@ partial class ApplicationEngine
 
         protected internal bool CheckSig(byte[] pubkey, byte[] signature)
         {
+            using MemoryStream ms = new MemoryStream();
```

Comment:
Why don't use `public static byte[] GetHashData(this IVerifiable verifiable, uint magic)`

Reply:
It is removed. Because the property `Hash` rely on the parameterless overload.

Reply:
But now the `GetHashData` doesn't return the same to sign

Reply:
Any idea?

Reply:
Maybe It could be easier to sign the hash

```
writer.Write(ProtocolSettings.Magic);
writer.Write(ScriptContainer.Hash);
```
?

Reply:
Currently, the hash is `sha256(sha256(data))`.

We can change it to:

```
hash = sha256(data)
signature = sign(sha265(magic + hash))
```

What do you think?

Reply:
It's good to me

Reply:
Fixed.